Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp

```diff
@@ -646,6 +651,106 @@ namespace AZ
         }
     }
 
+    void ComponentApplication::InitializeAllocatorSettings(int argc, char** argv)
+    {
+        constexpr AZStd::string_view StartupFileOption = "startup-cfg-file";
+        AZ::IO::FixedMaxPath startupCfgPath;
+
+        // First check if there is an "O3DE_STARTUP_CFG_FILE" environment variable set
+        // in the OS and use that as the startup cfg path
+        auto QueryStartupEnv = [](char* buffer, size_t size) -> size_t
+        {
+            // 1024 is an arbitrary power of 2 size that we already have
+            // a template instantiation of(so therefore the symbol size of the library will not increase).
+            // It should be more than enough to store any reasonable environment variable name that the caller
+            // would want to query
+            constexpr const char* StartupCfgEnvKey = "O3DE_STARTUP_CFG_FILE";
+            auto getEnvOutcome = AZ::Utils::GetEnv(AZStd::span(buffer, size), StartupCfgEnvKey);
+            return getEnvOutcome ? getEnvOutcome.GetValue().size() : 0;
```

Comment:
That is because an [AZ::IO::FixedMaxPath](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path_fwd.h#L38-L43) is backed by `fixed_string<1024>` string under the hood.